src/hooks.tsx

```diff
@@ -217,3 +217,15 @@ export const useIsInViewport = (
 
     return isInViewport;
 };
+
+export const useIsFirstRender = (): boolean => {
```

Comment:
you should use an state + effect, otherwise there won't be a re-render.
Something like this:
https://beta.reactjs.org/reference/react-dom/client/hydrateRoot#handling-different-client-and-server-content

Reply:
mmm in this case, do we want that behaviour here? `useIsFirstRender` is this case, is we want to force a re-render, maybe we need a new name? any idea?

also, we want to force a re-render for this animation? in the case of the stepper, that animation will take place once yo go to next/prev state, so no need of that state + effect, I have doubts in adding that state + effect in the case of the icons,... WDYT?